File: vaultService.py
import requests

url = "http://hashicorpvault.corp.multiscale.tech/v1/testsecret/data/"
token = "hvs.UdOmnFHEXyCVjkzYGRjmqb2r"
headers={"X-Vault-Token": token}

def getVaultValue(path,keys):
    val = []
    response = requests.get(url + path, headers=headers)
    if response.status_code == 200:
        vault_data = response.json()['data']['data']
        for key in keys:
            if key in vault_data:
                val.append(vault_data[key])
        return val
    else:
        print("Error file fetching response from vault, response code {}".format(response.status_code))
        raise Exception("Error file fetching response from vault, response code {}".format(response.status_code))

print(getVaultValue("mongo",["username","password"]))

File: vaultService.js
const axios = require("axios")

token="hvs.UdOmnFHEXyCVjkzYGRjmqb2r"
url="https://hashicorpvault.corp.multiscale.tech/v1/testsecret/data/"
const requestConfig = {
    headers: {
        "X-Vault-Token": token
    }
}

async function getVaultKeyValue(path, ...keys) {
    vaultPath = url + path;
    vals = new Array()
    const response = await axios.get(vaultPath, requestConfig);
    vaultData = response.data.data.data;
    for(let key of keys){
	if(key in vaultData) {
		vals.push(vaultData[key])
	}
    }
    return vals;
}

getVaultKeyValue("mysql", "username","password").then(value => console.log(value));

